src/libs/WorkflowUtils.ts

```diff
@@ -0,0 +1,162 @@
+     */
+    employees: PolicyEmployeeList;
+
+    /**
+     * Personal details of the employees
+     */
+    personalDetails: PersonalDetailsList;
+
+    /**
+     * Email of the default approver for the policy
+     */
+    defaultApprover: string;
+};
+
+/** Convert a list of policy employees to a list of approval workflows */
+function convertPolicyEmployeesToApprovalWorkflows({employees, defaultApprover, personalDetails}: ConvertPolicyEmployeesToApprovalWorkflowsParams): ApprovalWorkflow[] {
+    const approvalWorkflows: Record<string, ApprovalWorkflow> = {};
+
+    // Keep track of how many times each approver is used to detect approvers in multiple workflows
+    const approverCountsByEmail: Record<string, number> = {};
+    const personalDetailsByEmail = lodashMapKeys(personalDetails, (value, key) => value?.login ?? key);
+
+    // Add each employee to the appropriate workflow
+    Object.values(employees).forEach((employee) => {
+        const {email, submitsTo} = employee;
+        if (!email || !submitsTo) {
+            return;
+        }
+
+        const member: Member = {email, avatar: personalDetailsByEmail[email]?.avatar, displayName: personalDetailsByEmail[email]?.displayName ?? email};
+        if (!approvalWorkflows[submitsTo]) {
+            const approvers = getApprovalWorkflowApprovers({employees, firstEmail: submitsTo, personalDetailsByEmail});
+            approvers.forEach((approver) => (approverCountsByEmail[approver.email] = (approverCountsByEmail[approver.email] ?? 0) + 1));
+
+            approvalWorkflows[submitsTo] = {
+                members: [],
+                approvers,
+                isDefault: defaultApprover === submitsTo,
+                isBeingEdited: false,
+            };
+        }
+        approvalWorkflows[submitsTo].members.push(member);
+    });
+
+    // Sort the workflows by the first approver's name (default workflow has priority)
+    const sortedApprovalWorkflows = Object.values(approvalWorkflows).sort((a, b) => {
+        if (a.isDefault) {
+            return -1;
+        }
+
+        return (a.approvers[0]?.displayName ?? '-1').localeCompare(b.approvers[0]?.displayName ?? '-1');
+    });
+
+    // Add a flag to each approver to indicate if they are in multiple workflows
+    return sortedApprovalWorkflows.map((workflow) => ({
+        ...workflow,
+        approvers: workflow.approvers.map((approver) => ({
+            ...approver,
+            isInMultipleWorkflows: approverCountsByEmail[approver.email] > 1,
+        })),
+    }));
+}
+
+type ConvertApprovalWorkflowToPolicyEmployeesParams = {
+    approvalWorkflow: ApprovalWorkflow;
+    employeeList: PolicyEmployeeList;
+    removeWorkflow?: boolean;
+};
+
+/** Convert an approval workflow to a list of policy employees */
+function convertApprovalWorkflowToPolicyEmployees({approvalWorkflow, employeeList, removeWorkflow = false}: ConvertApprovalWorkflowToPolicyEmployeesParams): PolicyEmployeeList {
```

Comment:
I think the result of this method is in the wrong format, but it's close! There are two formats for policy employees:

- What is stored in Onyx only used on the frontend (an object keyed by email):

```
{
    '1@example.com': {email: '1@example.com'},
}
```

- What is stored in the database and the format that the backend users (an array of objects):

```
[
    {email: '1@example.com'},
]
```

I believe this method needs to produce the second format because that is the format that the backend API will be expecting. You can see this being done in the [API parameters for updating a workspace members role](https://github.com/Expensify/App/blob/d8c0a7338fe9402817e8a267b6a289aa98f7cfad/src/libs/actions/Policy/Member.ts#L469-L471) (this endpoint is also a direct alias for `Policy_Employees_Merge` command for OldDot.

Shouldn't be a big deal to change this and then update the tests to match for format. I don't _think_ we need to support converting workflows into the first employeeList format.

Reply:
Ah, I see in your second PR that you will need to use the first format for the optimistic data here: https://github.com/Expensify/App/pull/46189/files#diff-0322d176e073c6db8f9e2a5203bc59580b8e4cda517da2ccddf6d53318789ff3R45-R47

So it needs to support both formats.

Reply:
Or... I could also change the backend to accept the employeeList in the same format that the frontend is using 🤔

Reply:
I think converting to an array it's not a big deal (one call to `Object.values()`). Therefore, I don't think it's necessary to change the format on the backend.